### PR TITLE
Role support

### PR DESCRIPTION
**Pull request by brandon515**
Adds support for the creation, modification, and deletion of roles in a server, 3 functions are added and one new builder is created.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -259,6 +259,25 @@ impl Discord {
 		check_status(result)
 	}
 
+    /// Create a role.
+    pub fn create_role(&self, server: ServerId) -> Result<Role>{
+        let response = request!(self, post, "/guilds/{}/roles", server);
+        Role::decode(try!(serde_json::from_reader(response)))
+    }
```
Comment (ghost) [inline src/lib.rs:266]: Discord has released an [API update](https://github.com/hammerandchisel/discord-api-docs/commit/ee358f6f0870d3153f153222a2b62b1027f69536) to allow setting properties on create. The `EditRole` builder can be used here too now.

File: src/builders.rs [hunk 2]
```diff
@@ -188,6 +191,38 @@ impl EditUserProfile {
...
+
+    /// Bitwise of the enabled/disabled permissions.
+    pub fn permissions(self, perm: u64) -> Self{
+        EditRole(self.0.insert("permissions", perm))
+    }
+
+    /// Edit the sorting postion of the role.
+    pub fn position(self, pos: u64) -> Self{
+        EditRole(self.0.insert("position", pos))
+    }
```
Comment (ghost) [inline src/builders.rs:208]: The position can't be modified on a PATCH for a single role; this needs to be done in a [batch guild roles PATCH](https://discordapp.com/developers/docs/resources/guild#modify-guild-role-positions), i.e. a separate `Discord::move_roles` method.

File: src/builders.rs [hunk 3]
```diff
@@ -188,6 +191,38 @@ impl EditUserProfile {
...
+        EditRole(self.0.insert("color", color))
+    }
+
+    /// Whether the role should be displayed seperately in the sidebar
+    pub fn hoist(self, hoist: bool) -> Self{
+        EditRole(self.0.insert("hoist", hoist))
+    }
+
+    /// Where the role should be mentionable
+    pub fn mentionable(self, mentionable: bool) -> Self{
```
Comment (ghost) [inline src/builders.rs:221]: Probably a space after `Self` and before the curly brace, and for above

**Review by ghost — COMMENTED** [review]
Mostly leaving a comment due to now being able to set properties on role POST. Additionally, discord-rs uses tabs for indentation

**Comment by anna-is-cute** [comment]
What is keeping this or #100 from being merged?

**Comment by SpaceManiac** [comment]
I'm busy with school and other things and haven't had time to really sit and review them side-by-side. It will happen eventually.

**Comment by Tormyst** [comment]
If your interested, I did an update to the code from #100.  I don't know if I should create another pull request for this issue, but you are welcome to see it on my [fork](https://github.com/Tormyst/discord-rs).

**Comment by jD91mZM2** [comment]
Need this. Bump?

**Comment by anna-is-cute** [comment]
This project is dead. I suggest finding an alternate library such as [serenity](https://crates.io/crates/serenity). Frankly, I'm amazed this still compiles for you.

**Comment by jD91mZM2** [comment]
Ugh that sucks :(. Serenity has so much fancy stuff and I don't want the fancy stuffs :(

**Comment by anna-is-cute** [comment]
I'd wager you can turn off most of the ["fancy stuff"](https://github.com/serenity-rs/serenity/blob/master/Cargo.toml#L85) just by using `no-default-features`! :)

**Comment by jD91mZM2** [comment]
Oh I see. Wow, thanks!

**Comment by SpaceManiac** [comment]
Closed in favor of #100 and #131.

**Pull request closed without merging**